x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -653,15 +652,12 @@ private static List<EsIndex> buildIndices(
             }
         }
 
-        List<String> resolvedIndices = new ArrayList<>(asList(fieldCapsResponse.getIndices()));
-        int mapSize = CollectionUtils.mapSize(resolvedIndices.size() + resolvedAliases.size());
-        Map<String, Fields> indices = Maps.newLinkedHashMapWithExpectedSize(mapSize);
+        final List<String> resolvedIndices = arrayAsArrayList(fieldCapsResponse.getIndices());
+        Map<String, Fields> indices = Maps.newLinkedHashMapWithExpectedSize(resolvedIndices.size() + resolvedAliases.size());
```

Comment:
This was computing the size wrong, `newLInkedHashMapWithExpected` size already accounts for the load factor.

Reply:
The bug is not in `CollectionsUtils.mapSize` :), it's the fact that we apply both `CollectionsUtils.mapSize` **and then also** use `Map.capacity` in `newLinkedHashMapWithExpectedSize` so we oversize the map :)